### PR TITLE
Support Python 3.11

### PR DESCRIPTION
**Pull request by susodapop**
## Description

**Status**: This PR is on-hold awaiting PyArrow to release their pre-built wheels for Python 3.11.

This PR adds Python 3.11 to the unit testing matrix.

Closes https://github.com/databricks/databricks-sql-python/issues/59

### CODE REVIEW

**Review by moderakh — APPROVED** [review]

**Comment by susodapop** [comment]
The Python 3.11 tests fail because our numpy dependency is pinned to 1.21.1 but Python 3.11 support was added in 1.24.2.

I'm going to split the dependency specification so that 1.21.1 is installed for Python 3.7 and a newer version is used for newer Pythons. This dependency resolution takes a long time on my workstation though. Will update this branch once the resolution completes.

**Comment by susodapop** [comment]
Alright, I fixed the issue with numpy but now pyarrow is raising concerns 😬 . Will continue debugging and update this PR as I have more info.

**Comment by susodapop** [comment]
Looks like the Pyarrow team are at work on this here https://github.com/apache/arrow/pull/14499

**Comment by susodapop** [comment]
I tried manually installing `pyarrow` in Python 3.11 on my Intel Mac and it worked. Only _some_ of their wheels aren't building, including whatever GH actions needs to run its tests.

**Comment by susodapop** [comment]
I forced a push after rebasing on `main` which incorporates the new `thrift` dependency. I still expect the Python 3.11 build to fail because Arrow have not published wheels for Python v11 yet.

File: pyproject.toml [hunk 1]
```diff
@@ -14,8 +14,12 @@ thrift = "^0.16.0"
 pandas = "^1.3.0"
 pyarrow = "^9.0.0"
```
Comment (alexmalins) [inline pyproject.toml:15]: The Python 3.11 compatible wheels are for pyarrow versions >=10.0.1, so this line needs to change to make this PR work.

You could try using `pyarrow = ">=9.0.0"` or maybe better

```
pyarrow = [
    {version = ">=9.0.0", python = ">=3.7,<3.11"},
    {version = ">=10.0.1", python = ">=3.11"}
]
```

Comment (alexmalins) [reply pyproject.toml:15]: Hey, as @potiuk said in #59, the pyarrow wheels for Python 3.11 are now out.

I could install this package successfully from a local git repo clone by slacking off the above pyarrow requirement. Please see above comment

**Comment by susodapop** [comment]
I rebuilt the lockfile with our latest change including @alexmalins revised pyarrow dependency spec. I'm watching the PR checks now to make sure they go through. Assuming they do, we'll bump our version and cut a new release in the next 24 hours.

**Comment by susodapop** [comment]
Everything builds 🚀 

I ran the e2e tests from Python 3.11 and Python 3.10.5 and all pass from `PySQLCoreTestSuite`.

**Comment by potiuk** [comment]
Woooohooo ! I hope we get binary wheel in Pypi soon :)

**Comment by susodapop** [comment]
Version `2.2.1` with Python 3.11 support is deploying to Pypi now: https://github.com/databricks/databricks-sql-python/actions/runs/3569089528

**Comment by susodapop** [comment]
Update: v2.2.1 is now available on Pypi.

**Comment by potiuk** [comment]
Tomorrows news today :) . Thanks !

**Comment by susodapop** [comment]
Thanks for your help and persistence on this!

**Pull request merged**